compiler/rustc_mir/src/transform/coverage/debug.rs

```diff
@@ -148,23 +148,19 @@ impl DebugOptions {
 
         if let Ok(env_debug_options) = std::env::var(RUSTC_COVERAGE_DEBUG_OPTIONS) {
             for setting_str in env_debug_options.replace(" ", "").replace("-", "_").split(',') {
-                let mut setting = setting_str.splitn(2, '=');
-                match setting.next() {
```

Comment:
The original match was considering more than necessary. `setting.next()` was always `Some()`. Only the `value` could be `None`. 

I kept `value` modeled as an `Option` to avoid needing to change the related code like `bool_option_val`.